forch/faucetizer.py

```diff
@@ -61,7 +65,11 @@ def _faucetize(self):
                                    device.placement.switch, device.placement.port)
                     continue
 
-                port_cfg['native_vlan'] = device.behavior.vid
+                vid = self._segments_to_vlans.get(device.behavior.segment)
+                if not vid:
+                    raise Exception('Device segment does not have a matching vlan %s %s',
```

Comment:
this will bork the entire configuration, which is likely overkill.  I would log a warning (since it's not really an error in the system, but one of maybe how RADIUS is configured), and then punt the device to the initialization vlan.

Reply:
I would also exctact this to a function like self._get_segment_for_behavior(....) and encapsulate everything there

Reply:
Sorry, should be _get_vlan_for_behavior, so it's just:

vid = self._get_vlan_for_behavior(device.behavior)